### PR TITLE
tests: Ports kubelet unit tests to Windows

### PR DESCRIPTION
**Pull request by claudiubelu**
#### What type of PR is this?

/kind failing-test

/sig testing
/sig windows
/priority important-soon
/milestone v1.25

#### What this PR does / why we need it:

Ports kubelet/util unit tests to Windows.

#### Which issue(s) this PR fixes:

Related: https://github.com/kubernetes/kubernetes/issues/51540

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by harshitasao** [comment]
Hi @claudiubelu, k8s 1.25 Bug Triage Shadow here, just checking in to see if this is still on track for k8s 1.25?
Thank you.

**Comment by claudiubelu** [comment]
> Hi @claudiubelu, k8s 1.25 Bug Triage Shadow here, just checking in to see if this is still on track for k8s 1.25? Thank you.

Yes.

**Comment by harshitasao** [comment]
Hello @claudiubelu, with the code freeze approaching, is this still on track for v1.25?
Thanks!

**Comment by marosset** [comment]
> Hello @claudiubelu, with the code freeze approaching, is this still on track for v1.25? Thanks!

We hope to get this merged but need reviews from multiple areas/owners.

**Comment by marosset** [comment]
These new unit tests are passing in https://github.com/kubernetes/kubernetes/pull/110981

**Comment by helayoty** [comment]
@claudiubelu  @marosset @feiskyer 👋 the release 1.25 bug triage Lead. I'm reaching out to see the status of this PR and if we are still targeting the current release. Also, is it a release blocker?

**Comment by marosset** [comment]
This is not a release blocker
/milestore clear

File: pkg/kubelet/util/util_windows_test.go
Comment (SergeyKanzhelev) [inline pkg/kubelet/util/util_windows_test.go:221]: from a quick glance, I do not understand this change. The string still has two substitutions to make

Comment (claudiubelu) [reply pkg/kubelet/util/util_windows_test.go:221]: We've removed the ``label`` from these tests. The substitution for it remained, but it should have been removed. Thanks for catching this!

File: pkg/kubelet/util/util_windows_test.go
Comment (SergeyKanzhelev) [inline pkg/kubelet/util/util_windows_test.go:199]: `assert.NoError` would work better here?

File: pkg/kubelet/util/util_windows_test.go
Comment (SergeyKanzhelev) [inline pkg/kubelet/util/util_windows_test.go:197]: not sure why not `assert.NoError`?

Comment (SergeyKanzhelev) [reply pkg/kubelet/util/util_windows_test.go:197]: I understand it is not strictly related to this PR

**Review by SergeyKanzhelev — APPROVED** [review]
/lgtm

**Comment by valaparthvi** [comment]
Hi, I am from the Bug Triage Shadow Team for K8s release 1.27.
I see this PR is waiting for an approval from SIG Node. Can @kubernetes/sig-node-leads give this PR a review so that we can keep it on track for v1.27.

**Pull request merged**
